packages/trpc/server/routers/viewer/organizations/create.handler.ts

```diff
@@ -124,12 +151,252 @@ export const createHandler = async ({ input, ctx }: CreateOptions) => {
+    }));
+
+  const user = await UserRepository.enrichUserWithItsProfile({
+    user: { ...orgOwner, organizationId: id },
+  });
+
+  return {
+    userId: user.id,
+    email: user.email,
+    organizationId: user.organizationId,
+    upId: user.profile.upId,
+  };
+};
+
+type OrgData = {
+  name: string;
+  slug: string;
+  isOrganizationConfigured: boolean;
+  isOrganizationAdminReviewed: boolean;
+  autoAcceptEmail: string;
+  seats: number | null;
+  pricePerSeat: number | null;
+  isPlatform: boolean;
+};
+
+type PersistOrgWithExistingOwner = {
+  ownerId: User["id"];
+  ownerUsername: User["username"];
+  orgId: number | null;
+};
+
+type PersistOrganizationWithExistingUserAsOwnerProps = {
+  orgOwner: User;
+  orgData: OrgData;
+  orgOwnerEmail: string;
+  userName: string | null;
+  translation: TFunction;
+  loggedInUserId: number;
+  availability: Availability;
+  existingOwnerDetails: PersistOrgWithExistingOwner;
+};
+
+const persistOrganizationWithExistingUserAsOwner = async ({
+  orgData,
+  orgOwner,
+  orgOwnerEmail,
+  userName,
+  availability,
+  translation,
+  loggedInUserId,
+  existingOwnerDetails: { orgId, ownerId: orgOwnerId, ownerUsername },
+}: PersistOrganizationWithExistingUserAsOwnerProps) => {
+  const isLoggedInUserOrgOwner = orgOwnerId === loggedInUserId;
+  if (orgId && isLoggedInUserOrgOwner) {
+    throw new TRPCError({ code: "FORBIDDEN", message: "User is part of an organization already" });
+  }
+
+  const nonOrgUsernameForOwner = ownerUsername || "";
+  const { organization, ownerProfile } = await OrganizationRepository.createWithExistingUserAsOwner({
+    orgData,
+    owner: {
+      id: orgOwnerId,
+      email: orgOwnerEmail,
+      nonOrgUsername: nonOrgUsernameForOwner,
+    },
+  });
+
+  if (!organization.id) throw Error("User not created");
+
+  const user = await sendEmailAndEnrichProfile({
+    org: { owner: orgOwner, id: organization.id, name: organization.name, ownerEmail: orgOwnerEmail },
+    slug: orgData.slug,
+    userName: userName,
+    ownerProfileUsername: ownerProfile.username,
+    inputLanguageTranslation: translation,
+    nonOrgUsernameForOwner: nonOrgUsernameForOwner,
+    isPlatform: orgData.isPlatform,
+  });
+
+  await prisma.availability.createMany({
+    data: availability.map((schedule) => ({
+      days: schedule.days,
+      startTime: schedule.startTime,
+      endTime: schedule.endTime,
+      userId: user.userId,
+    })),
+  });
+
+  return user;
+};
+
+type PersistOrganizationWithNonExistentOwnerProps = {
+  orgData: OrgData;
+  orgOwner: User | null;
+  orgOwnerEmail: string;
+  userLocale: string;
+  userName: string | null;
+  translation: TFunction;
+};
+
+const persistOrganizationWithNonExistentOwner = async ({
+  orgData,
+  orgOwner,
+  orgOwnerEmail,
+  userLocale,
+  userName,
+  translation,
+}: PersistOrganizationWithNonExistentOwnerProps) => {
+  let organizationOwner = orgOwner;
```

Comment:
not sure to understand why we need to do this, organizationOwner is fully replaced with data.orgOwner, 

the orgOwner params does not seem to be used at all really